project/Build.scala

```diff
@@ -70,58 +116,60 @@ object DottyBuild extends Build {
       lockDir.mkdirs
       // Cannot have concurrent partests as they write to the same directory.
       if (lockDir.list.size > 0)
-        throw new RuntimeException("ERROR: sbt partest: another partest is already running, pid in lock file: " + lockDir.list.toList.mkString(" "))
+        throw new RuntimeException(
+          s"""|ERROR: sbt partest: another partest is already running,
+              |       pid in lock file: ${lockDir.list.mkString(" ")}""")
       partestLockFile.createNewFile
       partestLockFile.deleteOnExit
     },
+
     runPartestRunner <<= Def.inputTaskDyn {
       // Magic! This is both an input task and a dynamic task. Apparently
       // command line arguments get passed to the last task in an aliased
       // sequence (see partest alias below), so this works.
       val args = Def.spaceDelimited("<arg>").parsed
-      val jars = Seq((packageBin in Compile).value.getAbsolutePath) ++
-          getJarPaths(partestDeps.value, ivyPaths.value.ivyHome)
-      val dottyJars  = "-dottyJars " + (jars.length + 1) + " dotty.jar" + " " + jars.mkString(" ")
+      val dottyJars = Seq("dotty.jar",
+                     (packageBin in Compile).value.getAbsolutePath) ++
+                 getJarPaths(partestDeps.value, ivyPaths.value.ivyHome) ++
+                 Seq("dotty.jar")
+
+      val dottyJarsConf  =
+        List("-dottyJars", dottyJars.length, dottyJars) ++ args
```

Comment:
This should be `List("-dottyJars", dottyJars.length) ++ dottyJars ++ args`, otherwise you get a List in your Seq and `dottyJarsConf.mkString(" ")` is wrong, this is why the tests fail currently.